src/js/tracks/text-track.js

```diff
@@ -107,6 +107,11 @@ vjs.TextTrack = function(options) {
         return;
       }
       mode = newMode;
+
+      if (mode !== 'disabled' && this['cues'].length === 0) {
+	loadTrack(this.src_, this);
+      }
+
       if (mode === 'showing') {
```

Comment:
This should include hidden tracks too so cuechange events fire for those also. Would that mess anything up?